### PR TITLE
Auto Deletion

### PR DESCRIPTION
**Pull request by Dustin-Layne**
Added methods to auto delete and auto place blocks. Auto delete can be toggled with 'v' and auto place can be toggled with 'c'. These will automatically place blocks or auto delete blocks when available allowing the user to not have to spam click. They are best used when creating or digging out terrain on a mass scale.

These methods need tweaked to allow the user a little more control, mainly the auto placement, but this is a rudimentary build.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -2394,6 +2396,22 @@ void on_key(GLFWwindow *window, int key, int scancode, int action, int mods) {
                 g->isWalking = false;     
 			} 
 		}
+        if (key == CRAFT_KEY_AUTODELETE) {
```
Comment (jrichards1999) [inline src/main.c:2399]: Once again, great use of descriptive variables names for everything added.

File: src/main.c [hunk 2]
```diff
@@ -2394,6 +2396,22 @@ void on_key(GLFWwindow *window, int key, int scancode, int action, int mods) {
...
+            if(g->isDeleting) {
+                g->isDeleting = false;
+			} else {
+                g->isDeleting = true;
+                g->isPlacing = false;
+			}
+		}
+        if (key == CRAFT_KEY_AUTOPLACE) {
```
Comment (jrichards1999) [inline src/main.c:2407]: Each of these if statements here could benefit from some comments explaining what exactly happens when each of these keys is pressed.

**Review by jrichards1999 — COMMENTED** [review]
Great work Dustin.  I am a little unsure of what the code does when reading through the code, so comments are suggested, but the commit comment does a good job of explaining what each button now does.

**Comment by n0dam** [comment]
Nice work with literal programming. Could use some comments to be safe but they are not actually needed as this code with the variable names being clear does a nice job of explaining itself.

**Pull request merged**